lib/lz4.c

```diff
@@ -723,41 +794,76 @@ LZ4_FORCE_INLINE int LZ4_compress_generic(
         op += lastRun;
     }
 
+    retval = (((char*)op)-dest);
+
+_clean_up:
+    if (dictDirective == usingExtDictCtx) {
+      /* Subsequent linked blocks can't use the dictionary. */
+      /* Instead, they use the block we just compressed. */
+      cctx->dictCtx = NULL;
+      cctx->dictSize = (U32)inputSize;
+    } else {
+      cctx->dictSize += (U32)inputSize;
+    }
+    cctx->currentOffset += (U32)inputSize;
+    cctx->tableType = tableType;
+
     /* End */
-    return (int) (((char*)op)-dest);
+    return (int)retval;
 }
 
 
 int LZ4_compress_fast_extState(void* state, const char* source, char* dest, int inputSize, int maxOutputSize, int acceleration)
+{
+  LZ4_resetStream((LZ4_stream_t*)state);
+  return LZ4_compress_fast_safeExtState(state, source, dest, inputSize, maxOutputSize, acceleration);
+}
+
+
+int LZ4_compress_fast_safeExtState(void* state, const char* source, char* dest, int inputSize, int maxOutputSize, int acceleration)
```

Comment:
`_safeExtState()` ?
Can you explain the logic behind the name ?

Reply:
Note : I could read the associated code comment in `lz4.h`.
Note that you can copy/paste code comments from `.h` into `.c` when they make sense to explain a function objective. It's not required either, but it can make sense when it's not obvious.

In this case, the naming does not make the scope of this function obvious, so it needs a comment to explain what it does.

It's also possible to change the name of the function.

For example, `LZ4_compress_fast_noReset()` ? or `_noStateReset` ?